app/Http/Controllers/Core/ProductController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Http\Controllers\Core;
+
+
+use App\Core\Models\Product;
+use App\Core\Models\Provider;
+use App\Core\Services\ProductService;
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+
+/**
+ * Created by PhpStorm.
+ * User: KnightSaber77
+ * Date: 1/16/2018
+ * Time: 3:47 PM
+ */
+
+class ProductController extends Controller
+{
+    private $productService;
+    public function __construct()
+    {
+        $this->productService = new ProductService();
+    }
+
+    public function productShow()
+    {
+        $data['products'] = Product::all();
+        $data2['providers'] = Provider::all();
```

Comment:
ini juga ikutin yang atas ya, manggil providerService dulu (jangan lupa providerServicenya di definisikan di `__construct`

terus ini nggak perlu bikin `$data2` karena `$data` itu udah array,
jadi cukup:
`$data['providers'] = $this->providerService->getAll();`